TUTORIAL.md

```diff
@@ -1239,6 +1239,55 @@ each bit. `0b0000_1000` evaluates to decimal `8`
 The integer type can be specified as a suffix to the binary literal,
 so `0b0100u8` evaluates to decimal `4` as an unsigned 8-bit integer.
 
+## Crashing
+
+Ideally, Roc programs would never crash. However, there are some circumstances in which they may:
+
+1. When doing normal integer arithmetic (e.g. `x + y`) that overflows.
+2. When the system runs out of memory.
+3. When a variable-length collection (like a `List` or `Str`) gets too long to be representible in the operating system's address space. (A 64-bit operating system's address space can represent several [exibytes](https://en.wikipedia.org/wiki/Byte#Multiple-byte_units) of data, so this case should not come up often.)
+
+Crashes are not like [`try`/`catch` exceptions](https://en.wikipedia.org/wiki/Exception_handling)
+found in some other programming languages. There is no way to "catch" a crash. It immediately
+ends the program, and what happens next is defined by the platform. (For example, a command-line
+interface platform might exit with a nonzero [exit code](https://en.wikipedia.org/wiki/Exit_status),
+whereas a web server platform might have the current request respond with a
+[HTTP 500 error](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#500).)
+
+You can intentionally crash a Roc program, for example if you encounter a conditional branch that
+you believe is unreachable. For example, you might know for certain that a particular `List U8`
```

Comment:
This sentence is a little awkward since it has a for example clause, and the very next sentence also starts with “for example”. I know this is still in draft but just wanted to call it out when I saw it!